### DISCUSSION TITLE
Is this the right way to use BWS with variables?

### DISCUSSION THREAD

**Discussion by returntrip** [twpayne/chezmoi#3723] [Q&A] [answered]
I am trying to populate the "$email" value using Bitwarden secrets, in such way:

```
{{- $email :=  (bitwardenSecrets "something123").value -}}                                                                     
{{- if $work -}}                                                                                                                                       
{{-   $email =  (bitwardenSecrets "something124").value -}}                                                                    
{{- end -}}
```

Unfortunately the value returned in the is always blank

I have tested the template like so:

```
$chezmoi execute-template '{{- $email :=  (bitwardenSecrets "something").value -}}'
nothing is returned

$chezmoi execute-template '{{- (bitwardenSecrets "something").value -}}'
email@domain.com
```

**Comment by twpayne** [comment]
> I am trying to populate the "$email" value using Bitwarden secrets, in such way:
> 
> ```
> {{- $email :=  (bitwardenSecrets "something123").value -}}                                                                     
> {{- if $work -}}                                                                                                                                       
> {{-   $email =  (bitwardenSecrets "something124").value -}}                                                                    
> {{- end -}}
> ```
> 
> Unfortunately the value returned in the is always blank

This is because this template does not result in any output. It only sets the `$email` variable.



> I have tested the template like so:
> 
> ```
> $chezmoi execute-template '{{- $email :=  (bitwardenSecrets "something").value -}}'
> nothing is returned
> ```

Again, the template only creates the `$email` variable. It does not result in any output.


> ``` 
> $chezmoi execute-template '{{- (bitwardenSecrets "something").value -}}'
> email@domain.com
> ```

This has output, as expected.

**Reply by returntrip** [reply, marked as answer]
Update: I readded my changes and this time it seemed to have worked. I will do some more testing and close this if all is fine. In the meanwhile, thanks for the help!

Sorry I should have posted the whole template. This is the original template which populates the email and other dtaa correctly. I was assuming that I could populate the email value using BWS. But I am probably doing something wrong.

```
{{- $work := false -}}{{/* true if this machine is used for work related tasks */}}
{{- $personal := false -}}{{/* true if this machine is used for personal tasks */}}
{{- $tty := false -}}{{/* true if this machine is a terminal based */}}

{{/* detect WSL or SSH sessions */}}
{{- if or (env "SSH_TTY") (.chezmoi.kernel.osrelease | lower | contains "microsoft") -}}
{{-   $tty = true -}}
{{- end -}}

{{- if eq .chezmoi.hostname "A" "B" -}}
{{-   $personal = true -}}
{{- end -}}

{{- if $tty -}}
{{-   if or (eq .chezmoi.hostname "WIN") (eq .chezmoi.osRelease.id "ubuntu" "fedoraremixforwsl") -}}
{{-     $work = true -}}
{{-   else -}}
{{-   $personal = true -}}
{{-   end -}}
{{- end -}}

{{- $email := "personal@email.com" -}}
{{- $name := "returntrip" -}}
{{- if $work -}}
{{-   $email = "work@email.com" -}}
{{-   $name = "Stefano Figura" -}}      
{{- end -}}

[git]
    autoCommit = true

[data]
    work = {{ $work }}
    personal = {{ $personal }}
    email = {{ $email | quote }}
    name = {{ $name | quote }}
```

**Reply by twpayne** [reply]
Please post an example that allows someone else to reproduce the problem.

**Comment by returntrip** [comment]
This is a non issue. Closing as everything is working fine.
